### PR TITLE
chore(core): fallback for language query

### PR DESCRIPTION
**Pull request by UncleSamtoshi**
- use empty language when notifictions service returns an error
- add timeout to notification settings grpc call

### CODE REVIEW

File: core/api/src/domain/users/index.ts [hunk 1]
```diff
@@ -39,6 +39,8 @@ export const checkedToEmailAddress = (
   return emailAddress as EmailAddress
 }
 
+export const DefaultLanguage = "" as UserLanguageOrEmpty
```
Comment (bodymindarts) [inline core/api/src/domain/users/index.ts:42]: With the default being set the type `UserLanguageOrEmpty` becomes irrelevant. Perhaps you could find something semantically more meaningful. Or if it's too complicated to untangle we can leave it for another PR.

Looks good otherwise.

Comment (UncleSamtoshi) [reply core/api/src/domain/users/index.ts:42]: Not sure I fully understand. The `DefaultLanguage` is used as a fallback in the case of the error, but even in the case there is not an error the notifications service could still return an empty string implying that the mobile application should use the OS system default. It's definitely a little bit of a mess that is a relic of how the mobile application was handling the language.

Comment (UncleSamtoshi) [reply core/api/src/domain/users/index.ts:42]: I think it may make sense to save that refactoring for a separate pr.

File: core/api/src/services/notifications/index.ts [hunk 2]
```diff
@@ -65,6 +64,9 @@ export const NotificationsService = (): INotificationsService => {
       const response = await notificationsGrpc.getNotificationSettings(
         request,
         notificationsGrpc.notificationsMetadata,
+        {
```
Comment (dolcalmi) [inline core/api/src/services/notifications/index.ts:67]: :thinking: shouldn't we use something similar to https://github.com/GaloyMoney/galoy/blob/main/core/api/src/services/lnd/index.ts#L801 ?

Comment (UncleSamtoshi) [reply core/api/src/services/notifications/index.ts:67]: That would work too, but I don't see a problem with relying on grpc.

**Review by dolcalmi — APPROVED** [review]
